lib/parallel.js

```diff
@@ -0,0 +1,23 @@
+// execute an array of asynchronous functions in parallel
+// @param {Array} fns - an array of functions
+// @param {Function} done - callback(err, results)
+function parallel (fns, done) {
+  var pending = fns.length
+  var results = []
+
+  function each (i, err, result) {
+    results[i] = result
+
+    if (--pending === 0 || err) {
+      done(err, results)
```

Comment:
done can be called more than once.
Actually if all the functions returns an error it's called `fns.length` times.

Is this an intended behaviour?

Reply:
good catch

Reply:
How do you want to handle that?
Maybe passing a strict option like discussed here?
https://github.com/soyuka/pidusage/issues/43#issuecomment-372324865